src/main/java/org/opensearch/neuralsearch/query/HybridQueryBuilder.java

```diff
@@ -0,0 +1,274 @@
+ * collects score for each of those sub-query.
+ */
+@Log4j2
+@Getter
+@Setter
+@Accessors(chain = true, fluent = true)
+@NoArgsConstructor
+public final class HybridQueryBuilder extends AbstractQueryBuilder<HybridQueryBuilder> {
+    public static final String NAME = "hybrid";
+
+    private static final ParseField QUERIES_FIELD = new ParseField("queries");
+
+    private final List<QueryBuilder> queries = new ArrayList<>();
+
+    private String fieldName;
+
+    private static final int MAX_NUMBER_OF_SUB_QUERIES = 5;
+
+    public HybridQueryBuilder(StreamInput in) throws IOException {
+        super(in);
+        queries.addAll(readQueries(in));
+    }
+
+    /**
+     * Serialize this query object into input stream
+     * @param out stream that we'll be used for serialization
+     * @throws IOException
+     */
+    @Override
+    protected void doWriteTo(StreamOutput out) throws IOException {
+        writeQueries(out, queries);
+    }
+
+    /**
+     * Add one sub-query
+     * @param queryBuilder
+     * @return
+     */
+    public HybridQueryBuilder add(QueryBuilder queryBuilder) {
+        if (queryBuilder == null) {
+            throw new IllegalArgumentException(String.format(Locale.ROOT, "inner %s query clause cannot be null", NAME));
+        }
+        queries.add(queryBuilder);
+        return this;
+    }
+
+    /**
+     * Create builder object with a content of this hybrid query
+     * @param builder
+     * @param params
+     * @throws IOException
+     */
+    @Override
+    protected void doXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject(NAME);
+        builder.startArray(QUERIES_FIELD.getPreferredName());
+        for (QueryBuilder queryBuilder : queries) {
+            queryBuilder.toXContent(builder, params);
+        }
+        builder.endArray();
+        printBoostAndQueryName(builder);
+        builder.endObject();
+    }
+
+    /**
+     * Create query object for current hybrid query using shard context
+     * @param queryShardContext context object that used to create hybrid query
+     * @return hybrid query object
+     * @throws IOException
+     */
+    @Override
+    protected Query doToQuery(QueryShardContext queryShardContext) throws IOException {
+        Collection<Query> queryCollection = toQueries(queries, queryShardContext);
+        if (queryCollection.isEmpty()) {
+            return Queries.newMatchNoDocsQuery(String.format(Locale.ROOT, "no clauses for %s query", NAME));
+        }
+        return new HybridQuery(queryCollection);
+    }
+
+    /**
+     * Creates HybridQueryBuilder from xContent.
+     * @param parser parser that has been initialized with the query content
+     * @return new instance of HybridQueryBuilder
+     * @throws IOException
+     */
+    public static HybridQueryBuilder fromXContent(XContentParser parser) throws IOException {
```

Comment:
lets add the query json over this to provide what we are exactly parsing.

Reply:
ack